detectable/src/test/java/com/synopsys/integration/detectable/detectables/setuptools/unit/SetupToolsExtractUtilsTest.java

```diff
@@ -0,0 +1,126 @@
+package com.synopsys.integration.detectable.detectables.setuptools.unit;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+
+import org.junit.jupiter.api.Test;
+import org.tomlj.TomlParseResult;
+
+import com.synopsys.integration.common.util.finder.FileFinder;
+import com.synopsys.integration.detectable.DetectableEnvironment;
+import com.synopsys.integration.detectable.detectables.setuptools.SetupToolsExtractUtils;
+import com.synopsys.integration.detectable.detectables.setuptools.parse.SetupToolsCfgParser;
+import com.synopsys.integration.detectable.detectables.setuptools.parse.SetupToolsParser;
+import com.synopsys.integration.detectable.detectables.setuptools.parse.SetupToolsPyParser;
+import com.synopsys.integration.detectable.detectables.setuptools.parse.SetupToolsTomlParser;
+
+public class SetupToolsExtractUtilsTest {
+    
+    private static final String TOML = ".toml";
+    private static final String BUILD_SYSTEM_REQUIRES_SETUPTOOLS = "[build-system]\nrequires = [\"setuptools\"]";
+    private static final String PYPROJECT = "pyproject";
+
+    @Test
+    public void testExtractToml() throws IOException {
+        String tomlContent = BUILD_SYSTEM_REQUIRES_SETUPTOOLS;
+        Path tempFile = Files.createTempFile(PYPROJECT, TOML);
+        Files.write(tempFile, tomlContent.getBytes());
+
+        TomlParseResult result = SetupToolsExtractUtils.extractToml(tempFile.toFile());
+
+        assertEquals("setuptools", result.getArray("build-system.requires").getString(0));
+
+        Files.delete(tempFile); // delete the temporary file
+    }
+
+    @Test
+    public void testCheckTomlRequiresSetupTools() throws IOException {
+        String tomlContent = BUILD_SYSTEM_REQUIRES_SETUPTOOLS;
+        Path tempFile = Files.createTempFile(PYPROJECT, TOML);
+        Files.write(tempFile, tomlContent.getBytes());
+
+        TomlParseResult result = SetupToolsExtractUtils.extractToml(tempFile.toFile());
+
+        assertTrue(SetupToolsExtractUtils.checkTomlRequiresSetupTools(result));
+
+        Files.delete(tempFile); // delete the temporary file
+    }
+    
+    @Test
+    public void testFindDependenciesFileForPyProjectToml() throws IOException {
+        String tomlContent = "[project]\ndependencies = [\"requests\"]";
+        Path tempFile = Files.createTempFile(PYPROJECT, TOML);
+        Files.write(tempFile, tomlContent.getBytes());
```

Comment:
I suppose this file is not expected to be terribly large to implement a buffered write.

Reply:
Yeah this piece is just a test so it's small for sure. I see your point though. Still, the projects we have been evaluating don't make me too concerned about this. I'm going to leave it this way for now and we can enhance as required.